File: Project/Simulator/Organisms/Plants/WolfBerries.java
package Simulator.Organisms.Plants;

import Simulator.Organisms.Organism;

import org.json.JSONObject;

import Simulator.World;

public class WolfBerries extends Plant {

    public WolfBerries(World world) {
        super(99, world, Type.WOLF_BERRIES);
    }

    public WolfBerries(JSONObject json, World world) {
        super(json, world);
    }

    @Override
    public Plant construct() {
        return new WolfBerries(world);
    }

    @Override
    public boolean collisionReaction(Organism other) {
        other.die();
        die();
        world.addLog(other.getSymbol() + " ate " + getSymbol() + " and died!");
        return true;
    }
}

File: Project/Simulator/Organisms/Plants/WolfBerries.py
from .Plant import Plant
from Simulator.Organisms.Organism import Type, Organism


class WolfBerries(Plant):
    def __init__(self, world, json=None):
        super().__init__(99, world, Type.WOLF_BERRIES, json)

    def construct(self):
        return WolfBerries(self.world)

    def collisionReaction(self, other: Organism):
        other.die()
        self.die()
        self.world.addLog(
            f"{other.getSymbol()} ate {self.getSymbol()} and died!")
        return True
